src/Analysis/Engine/Impl/ModuleAnalysis.cs

```diff
@@ -229,14 +229,14 @@ private class DefinitionWalker : PythonWalkerWithLocation {
             }
 
             if (expr is NameExpression name) {
-                var defScope = scope.EnumerateTowardsGlobal.FirstOrDefault(s =>
-                    s.ContainsVariable(name.Name) && (s == scope || s.VisibleToChildren || IsFirstLineOfFunction(scope, s, location)));
-
-                if (defScope == null) {
-                    variables = _unit.State.BuiltinModule.GetDefinitions(name.Name)
-                        .SelectMany(ToVariables);
+                if (!scope.EnumerateTowardsGlobal.Any()) {
```

Comment:
We don't need to call `EnumerateTowardsGlobal.Any()`, it should be enough to check after the `foreach` if `variables` is empty.